feat: add contribution instructions for Resoto (#8)
someengineering/resoto.com: docs/contributing/components.md

You will first need to [fork](https://docs.github.com/get-started/quickstart/fork-a-repo) the repository.

#### Prerequisites

- [Git](https://git-scm.com)
- [Python](https://python.org) 3.8+ (3.10 is recommended)
- [ArangoDB](https://arangodb.com) 3.8.2+
- [GNU Compiler Collection (GCC)](https://gcc.gnu.org) (depending on the host system, Python dependencies may need to be compiled from source)

#### Setting Up a Virtual Environment

We recommend using a [Python virtual environment](https://docs.python.org/3/tutorial/venv.html).

A script is provided to simplify the process of configuring the virtual environment:

```bash
./setup_venv.sh --dev --path .
```

Activate the virtual environment:

```mdx-code-block
<Tabs>
  <TabItem value="linux" label="Linux/macOS">
    <CodeBlock language="bash">source venv/bin/activate</CodeBlock>
  </TabItem>
  <TabItem value="windows" label="Windows">
    <CodeBlock language="powershell">venv\Scripts\activate.bat</CodeBlock>
  </TabItem>
</Tabs>
```

#### Starting the Database

Start ArangoDB (using `systemctl` on Linux, by clicking the application icon in macOS, etc.).

> **The expected default access is the built-in user account `root` with a configured empty string as password or authentication turned off.**
> This is necessary for `resotocore` to perform the database setup, and also required for tests to pass.

Reviewer: ```suggestion
Start ArangoDB (using `systemctl` on Linux, by clicking the application icon in macOS, etc.).

**By default, the built-in user account `root` does not have a password configured (i.e., the password defaults to an empty string).** This is necessary for `resotocore` to perform the database setup, and also required for tests to pass.
```
The main issue was "the expected default access," which sounds odd and is vague/unclear.  But I added what you have about the empty string!

Author: This sounds like this would be the default of the database.
That might or might not be the case.
It is important, that the user either already has this setup or configures the database this way:
- option 1: username: root, password: <empty string>
- option 2: turn authentication off

It does not seem very clear to me.

Reviewer: The phrase, "by default" means if the user did not do any prior configuration.  From a quick search, it seems that this is in fact the default for new ArangoDB installs.

Reviewer: But, we do need to address the case where users may have already configured ArangoDB.  I'll update my suggestion shortly.

Author: Yes. I installed arango on Fedora today - I needed to manually configure it.
I think it makes sense to say what we expect.
Would be great if you could make a proposal.

Reviewer: @aquamatthias Yup, I wasn't aware that the default password could vary depending on the install method.  Updated my suggestion above!

Author: 👍 thanks

## Authoring Changes

Contributions are made via [pull requests to the GitHub repository](https://github.com/someengineering/resoto/pulls).
You will first need to [fork](https://docs.github.com/get-started/quickstart/fork-a-repo) the repository.

#### Prerequisites

- [Git](https://git-scm.com)
- [Python](https://python.org) 3.8+ (3.10 is recommended)
- [ArangoDB](https://arangodb.com) 3.8.2+
- [GNU Compiler Collection (GCC)](https://gcc.gnu.org) (depending on the host system, Python dependencies may need to be compiled from source)

#### Setting Up a Virtual Environment

We recommend using a [Python virtual environment](https://docs.python.org/3/tutorial/venv.html).

A script is provided to simplify the process of configuring the virtual environment:

```bash
./setup_venv.sh --dev --path .
```

Activate the virtual environment:

```mdx-code-block
<Tabs>
  <TabItem value="linux" label="Linux/macOS">
    <CodeBlock language="bash">source venv/bin/activate</CodeBlock>
  </TabItem>
  <TabItem value="windows" label="Windows">
    <CodeBlock language="powershell">venv\Scripts\activate.bat</CodeBlock>
  </TabItem>
</Tabs>
```

#### Starting the Database

> After you have installed ArangoDB, you need to start it.
> The way the database is started depends on the way it has been installed (`systemctl` on Linux, clicking the application icon on macOS, etc.).
> 
> **The expected default access is the built-in user account `root` with a configured empty string as password.**
> No additional settings are required in such a setup, since `resotocore` will use this access to create all necessary items.
> This setting is required for tests to pass, as the tests create users and databases.

Reviewer: ```suggestion
Start ArangoDB (using `systemctl` on Linux, by clicking the application icon in macOS, etc.).

**By default, the built-in user account `root` does not have a password configured.** This is necessary for `resotocore` to perform the database setup, and also required for tests to pass.
```
Good to merge after this!

Author: ah sorry - the change was not committed. one sec
